### ISSUE TITLE
Comparing Versions with Pre-Release fail

### ISSUE THREAD

**Issue by raubv0gel** [PHLAK/SemVer#29]
The test
```php
public function test_failed(): void
{
    $version = new SemVer\Version('v1.3.37+build.1');
    $this->assertTrue($version->eq(new SemVer\Version('v1.3.37')));
}
```
**fails**. According to https://semver.org/#spec-item-10 `v1.3.37+build.1` and `v1.3.37` have to be equal.

The reason is comparing the pre-release part `null` with `''`:
![grafik](https://user-images.githubusercontent.com/9824146/221717900-431a4adf-d60b-4179-88ce-f40eadd8c788.png)

**Comment by PHLAK** [comment]
Good catch, thank you.

**Comment by PHLAK** [comment]
Fixed with 1816dfb119207c7cde08ae32e57d1ab353614a81
